ambry-cloud/src/main/java/com.github.ambry.cloud/StaticVcrClusterFactory.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud;
+
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.VirtualReplicatorCluster;
+import com.github.ambry.clustermap.VirtualReplicatorClusterFactory;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.ClusterMapConfig;
+
+
+/**
+ * {@link StaticVcrClusterFactory} to generate VCR Cluster for static partition assignment.
+ */
+public class StaticVcrClusterFactory implements VirtualReplicatorClusterFactory {
+
+  private final CloudConfig cloudConfig;
+  private final ClusterMapConfig clusterMapConfig;
+  private final ClusterMap clusterMap;
+  private VirtualReplicatorCluster virtualReplicatorCluster;
+
+  public StaticVcrClusterFactory(CloudConfig cloudConfig, ClusterMapConfig clusterMapConfig, ClusterMap clusterMap) {
+    this.cloudConfig = cloudConfig;
+    this.clusterMapConfig = clusterMapConfig;
+    this.clusterMap = clusterMap;
+  }
+
+  @Override
+  public VirtualReplicatorCluster getVirtualReplicatorCluster() {
```

Comment:
Singleton isn't threadsafe.  We're not worried about multiple threads calling this method?

Reply:
Good point.
So far, we don't have multiple thread use case here. It initialized once in VCRServer(). 
Also, StaticVcrClusterFactory is a temp solution for experiment. We will move to HelixVcrClusterFactory soon.

Reply:
Will `HelixVcrClusterFactory` access this in a single thread context?  

This page has a good design suggestion for how to write this in an efficient and thread-safe manner:

https://www.journaldev.com/171/thread-safety-in-java-singleton-classes-with-example-code
![image](https://user-images.githubusercontent.com/31049200/55655400-fb139e80-57a8-11e9-9d9b-3dcd6b133839.png)

Reply:
not sure if the `mutex` is necessary though as I think `synchronized` in a static context locks on the class object anyway

Reply:
`synchronized` added since this is a one time call. Will improve it if complex usages come. 
Thanks for the article.

Reply:
Another approach is to create the instance upfront in the constructor (eager vs lazy).  Then there is no need for synchronization.  Lazy construction is preferred if the object is heavyweight and you don't know ahead of time whether it will be needed.  But this getter will not be called many times.